### PR TITLE
nuclear: 0.6.6 -> 0.6.17

### PR DESCRIPTION
**Pull request by jakobneufeld**
###### Motivation for this change
I updated the nuclear audio player.
I like it.

###### Things done
Changed version.
Changed hash.
Build works on my nixos x64 machine.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jakobneufeld** [comment]
Question @IvarWithoutBones, are you the maintainer for this pakcage? It looks like that in the default.nix

**Comment by jakobneufeld** [comment]
Yay!!, the tests passed!

**Comment by danilevy1212** [comment]
Hi @jakobneufeld! I was recently using this package, but I can't get it to work, it opens just fine but it never reproduces any music I select. I addition, I can get it to change the directory to download music in:

Error when trying to play a song:
```shell
      main › renderer › Error while searching  for The Caretaker All you are going to want to do is get back there on Youtube
      main › renderer › Error: Error parsing info: Unexpected token ) in JSON at position 0
    at t.getBasicInfo (file:///nix/store/a1r7x3mf10dffvz4cy87lgvixfvbgab3-nuclear-v0.6.6-extracted/resources/app.asar/dist/1.renderer.js:2240:278393)
    at async Map.getOrSet (file:///nix/store/a1r7x3mf10dffvz4cy87lgvixfvbgab3-nuclear-v0.6.6-extracted/resources/app.asar/dist/1.renderer.js:2240:262532)
    at async t.getInfo (file:///nix/store/a1r7x3mf10dffvz4cy87lgvixfvbgab3-nuclear-v0.6.6-extracted/resources/app.asar/dist/1.renderer.js:2240:280578)
    at async Map.getOrSet (file:///nix/store/a1r7x3mf10dffvz4cy87lgvixfvbgab3-nuclear-v0.6.6-extracted/resources/app.asar/dist/1.renderer.js:2240:262532)
```

Error when opening the file browser:
```
(nuclear:2): GLib-GIO-ERROR **: 11:05:34.566: No GSettings schemas are installed on the system
```

Was wondering  if you had encountered these issues? I used your package recipe too but it's the same problem.

**Comment by jakobneufeld** [comment]
This is a weird error. I will try to reproduce on my machine. I will tell everyone my results

**Comment by IvarWithoutBones** [comment]
> Question @IvarWithoutBones, are you the maintainer for this pakcage? It looks like that in the default.nix

I am!

> Error when opening the file browser:
> `(nuclear:2): GLib-GIO-ERROR **: 11:05:34.566: No GSettings schemas are installed on the system`

This should be fixable by adding `wrapGAppsHook` to `nativeBuildInputs`. @jakobneufeld would you mind adding that?

Not entirely sure about the error above though.

**Comment by jakobneufeld** [comment]
I will add that

**Comment by jakobneufeld** [comment]
@IvarWithoutBones, is that what you meant?

**Comment by jakobneufeld** [comment]
I can not play any music, it just complains.

**Comment by danilevy1212** [comment]
Adding wrapGAppsHook to the nativeBuildInputs does NOT fix the issue for me.

**Comment by danilevy1212** [comment]
@jakobneufeld Me neither, the error that I show are what happens when I click play on a song.

**Comment by jakobneufeld** [comment]
Maybe, we should talk with upstream nuclear js. Maybe  some dependencies are missing. The error is cryptic. I have to go, but when I come back, I am happy to continue working on this issue.

**Comment by danilevy1212** [comment]
Yeah, their docs are not very descriptive either. Maybe it's best if we base the binary out of on of the Arch recipes instead of App Image. As far as I know, the app image should be plug and play, with all needed dependencies out of the box, which doesn't seem to be the case/ Another alternative is to try to pack the Flatpak/

**Comment by jakobneufeld** [comment]
@danilevy1212 How do we use the arch recepie. is there are pkg.aurToNix or something like that. Flatpak is good option too.

**Comment by jakobneufeld** [comment]
My experience is that when you talk with package maintainers, they just complain.

**Comment by danilevy1212** [comment]
@jakobneufeld About the AUR package. I don't there is any tool like that. What I meant is going through the package definition, pinning the dependencies in arch and changing them to the analog ones in nixpkgs and translating it's install scripts to nixpkgs.
It's a bunch of work, should be last resort if the flatpak doesn't work either :/

**Comment by jakobneufeld** [comment]
i am testing the flatpak now

**Comment by jakobneufeld** [comment]
can you try the flatpak @danilevy1212, because my audio setup is wierd and many audio applications do no support it. Here is the flatpak link if you need it https://flathub.org/apps/details/org.js.nuclear.Nuclear

**Comment by jakobneufeld** [comment]
OK, I tested the flatpak my self, and the app works fine, except I get this wierd error

**Comment by jakobneufeld** [comment]
image

**Comment by jakobneufeld** [comment]
I will report to upstream nuclear js

**Comment by jakobneufeld** [comment]
Here is the upstream issue
https://github.com/nukeop/nuclear/issues/1178

**Comment by danilevy1212** [comment]
Thanks so much @jakobneufeld, should we open another issue upstream for the AppImage too?

**Comment by WolfangAukang** [comment]
So curious, I've found about this Music Player today :).

FYI, @jakobneufeld your GitHub issue is the same as https://github.com/nukeop/nuclear/issues/1081. According to the developer, if you build master, it should work (they haven't thrown the release with the fix yet). I am going to test it right now, because as you said, even though I get that error message, I can get the Flatpak to work and download music, but not the NixOS version.

**Comment by jakobneufeld** [comment]
We have 3 options
1. Fix the app image with upstream (lots of work)
2.  Use flatpak, and ask upstream to build flatpak with master. 
3.  We use the AUR build script and package it ourselves (lots lots lots of work)

**Comment by WolfangAukang** [comment]
I think the nightly release already builds it with the most recent content. I'm testing right now.

**Comment by jakobneufeld** [comment]
Is there are nightly flatpak. Flatpak is the best option for this package

**Comment by WolfangAukang** [comment]
It seems not. I am not sure how you can download a flatpak file, but Flathub does not offer it on the nightly assets, so the only option would be building it from source.

I have built this app from source and although the error message has disappeared, I get the following error when playing something:
```
·> ./result/bin/nuclear
      main › (node:2) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `nuclear --trace-deprecation ...` to show where the warning was created)
  renderer › Cannot read property 'sourceName' of undefined
      main › (node:2) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'write' of null
    at e.exports.send (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/home/runner/work/nuclear/nuclear/node_modules/discord-rpc/src/transports/ipc.js:155:17)
    at /nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/home/runner/work/nuclear/nuclear/node_modules/discord-rpc/src/client.js:166:22
    at new Promise (<anonymous>)
    at e.exports.request (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/home/runner/work/nuclear/nuclear/node_modules/discord-rpc/src/client.js:164:12)
    at e.exports.clearActivity (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/home/runner/work/nuclear/nuclear/node_modules/discord-rpc/src/client.js:550:17)
    at hn.clear (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/src/services/discord/index.ts:97:16)
    at da.name [as onClearTrackList] (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/src/controllers/player.ts:63:18)
    at IpcMainImpl.<anonymous> (/nix/store/sdmfajqr1rg4r43l2k05bhbz2c7niw20-nuclear-nightly-extracted/resources/app.asar/dist/webpack:/src/utils/container.ts:54:39)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:9715)
    at Object.emit (events.js:315:20)
      main › (node:2) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
      main › (node:2) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
      main › error trying to set discord activity
```

**Comment by jakobneufeld** [comment]
You can dowload a flatpak ref file. Also keep in mind the I think that nuclear is in alpha. We need to link this issue with upstream if we want to continue. I appreciate that so many people came to this issue. I thought, this would be a simple package update :)

**Comment by jakobneufeld** [comment]
OK, I think the only way to continue, is that we build the package from master and then package that. The release  0.6.17 is to old for good NixOS use. Either we wait for a new release 0.6.18 which has all necessary bug fixes to get it working, or we build for master. How should we all continue?

**Comment by danilevy1212** [comment]
@jakobneufeld There is a nightly release for the AppImage https://github.com/nukeop/nuclear/releases

**Comment by jakobneufeld** [comment]
I am testing the nightly appimage

**Comment by danilevy1212** [comment]
I tested it out, but it's the same for me

**Comment by danilevy1212** [comment]
The AUR recipe is just the .deb file + some dependencies, maybe we could try that too:
- [AUR recipe](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=nuclear-player-bin)
- [NixOS packaging .deb files](https://unix.stackexchange.com/a/523174)

**Comment by WolfangAukang** [comment]
> @jakobneufeld There is a nightly release for the AppImage https://github.com/nukeop/nuclear/releases, from 3 days ago, maybe we could see if that one works?

This is the one I tried. It didn't work because of the error I stated before.

> The AUR recipe is just the .deb file + some dependencies, maybe we could try that too:

Already tried it. Complains about missing sqlite.

**Comment by WolfangAukang** [comment]
> OK, I think the only way to continue, is that we build the package from master and then package that. The release 0.6.17 is to old for good NixOS use. Either we wait for a new release 0.6.18 which has all necessary bug fixes to get it working, or we build for master. How should we all continue?

I would say having a nuclear nightly separate from the stable one.

**Comment by jakobneufeld** [comment]
OK, I think we should do two things.
1. Orphan or mark out of date this package
2. Make nuclear nightly like @WolfangAukang proposed.
Please thumps up to this comment.
Thank severyone for being so kind.
@everyone

**Comment by WolfangAukang** [comment]
FYI, I think we might need to wait before setting up the nightly. I just tried the new AppImage, and although the issue I had before has been fixed, I still don't have any sound playing when I click on a song.

**Comment by jakobneufeld** [comment]
Yeah, lets just wait for the next release 0.6.18

**Comment by jakobneufeld** [comment]
Nuclear music player is still in alpha I think so that is to be expected

File: pkgs/applications/audio/nuclear/default.nix [hunk 1]
```diff
@@ -21,7 +20,7 @@ in appimageTools.wrapType2 {
       --replace 'Exec=AppRun' 'Exec=${pname}'
     cp -r ${appimageContents}/usr/share/icons $out/share
   '';
-
+  extraPkgs = pkgs: with pkgs; [ wrapGAppsHook ];
```
Comment (Artturin) [inline pkgs/applications/audio/nuclear/default.nix:23]: this won't be needed once https://github.com/NixOS/nixpkgs/pull/161739 is merged

**Comment by WolfangAukang** [comment]
Hey everyone, in the end, there weren't any new releases, so I created a new PR (#169347) with the latest "pre-release" available (I say pre-release because all releases are shown as pre. The only difference is that the newest ones are created by GitHub Actions).

**Pull request closed without merging**